src/d/actor/d_a_kita.cpp

```diff
@@ -4,52 +4,448 @@
+                    i_this->field_2E4[i]++;
+                    shand_count++;
+                }
+            break;
+
+            case 2:
+            break;
+        }
+    }
+
+    return (shand_count < 4);
 }
 
 /* 000019F8-00001CB8       .text daKita_Execute__FP10kita_class */
-static BOOL daKita_Execute(kita_class*) {
-    /* Nonmatching */
+static BOOL daKita_Execute(kita_class* i_this) {
+    /* Nonmatching - static variables does not have the right address in .data section */
+    static float xd[4] = {130, -130, 130, -130};
+    static float zd[4] = {130, 130, -130, -130};
+
+    if(i_this->mExecuteCount != 0) i_this->mExecuteCount--;
+    kita_move(i_this);
+
+    MtxTrans(i_this->current.pos.x, i_this->current.pos.y, i_this->current.pos.z, false);
+    mDoMtx_YrotM(*calc_mtx, i_this->shape_angle.y);
+    mDoMtx_YrotM(*calc_mtx, i_this->mRotY);
+    MtxTrans(0.0f, 0.0f, (REG0_F(11) + -150.0f) * JMASSin(i_this->mRotX), true);
+    mDoMtx_XrotM(*calc_mtx, i_this->mRotX);
+    mDoMtx_YrotM(*calc_mtx, -i_this->mRotY);
+    mDoMtx_XrotM(*calc_mtx, i_this->shape_angle.x);
+    mDoMtx_ZrotM(*calc_mtx, i_this->shape_angle.z);
+    i_this->mModel->setBaseTRMtx(*calc_mtx);
+
+    cXyz local48[1];
+    for(int i = 0; i < 4; i++){
+        MtxPush();
+        local48->x = i_this->scale.x * xd[i];
+        local48->y = REG0_F(5) + -15.0f;
+        local48->z = i_this->scale.z * zd[i];
+        MtxPosition(local48, &i_this->field_2E8[i]);
+        MtxPull();
+    }
+    mDoMtx_copy(*calc_mtx, i_this->mBgwMtx);
+    i_this->pm_bgw->Move();
+    if(i_this->field_360 == 2){
+        if(i_this->mBaseEmitter == NULL){
+            i_this->mBaseEmitter = dComIfGp_particle_set(0x828D, &i_this->current.pos);
+        }
+        else {
+            i_this->mBaseEmitter->setGlobalTranslation(i_this->current.pos.x, i_this->current.pos.y - (REG0_F(17) + 40.0f), i_this->current.pos.z);
+        }
+    }
+    else {
+        if(i_this->mBaseEmitter != NULL){
+            i_this->mBaseEmitter->becomeInvalidEmitter();
+            i_this->mBaseEmitter = NULL;
+        }
+    }
+
+    return TRUE;
 }
 
 /* 00001CB8-00001CC0       .text daKita_IsDelete__FP10kita_class */
 static BOOL daKita_IsDelete(kita_class*) {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00001CC0-00001D3C       .text daKita_Delete__FP10kita_class */
-static BOOL daKita_Delete(kita_class*) {
-    /* Nonmatching */
+static BOOL daKita_Delete(kita_class* i_this) {
+    dComIfG_resDelete(&i_this->mPhs, "Kita");
+    if(i_this->heap != NULL){
+        dComIfG_Bgsp()->Release(i_this->pm_bgw);
+    }
+    if(i_this->mBaseEmitter != NULL){
+        i_this->mBaseEmitter->becomeInvalidEmitter();
+    }
+
+    return TRUE;
 }
 
 /* 00001D3C-00001EB0       .text CallbackCreateHeap__FP10fopAc_ac_c */
-static BOOL CallbackCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CallbackCreateHeap(fopAc_ac_c* p_this) {
+    BOOL ret;
+    kita_class* actor = static_cast<kita_class*>(p_this);
```

Comment:
Official names
```suggestion
static BOOL CallbackCreateHeap(fopAc_ac_c* i_this) {
    BOOL ret;
    kita_class* a_this = static_cast<kita_class*>(p_this);
```

Reply:
Oh interesting, I missed that when looking through the asserts, yeah that's fine then